pkg/dockerauth/interface.go

```diff
@@ -0,0 +1,7 @@
+package dockerauth
+
+import "github.com/fsouza/go-dockerclient"
+
+type AuthProvider interface {
+	GetAuthConfiguration(registry string) (docker.AuthConfiguration, error)
```

Comment:
We tend to not prefix method calls with `Get`. Can you change this method to just `AuthConfiguration`?